docs/docs/api/golang-bindings/lattice/pqc-ml-kem.md

```diff
@@ -0,0 +1,266 @@
+:::
+
+## Overview
+
+The `mlkem` package provides Golang bindings for Icicle's batched **ML-KEM** implementation (Kyber). It exposes three main functions – **`Keygen`**, **`Encapsulate`** and **`Decapsulate`** – that work on host or device memory.
+
+---
+
+## Public API
+
+### Key pair generation
+
+```go
+func Keygen(
+    params *KyberParams,
+    entropy core.HostOrDeviceSlice,  // batch_size × 64 bytes
+    config MlKemConfig,
+    publicKeys core.HostOrDeviceSlice,  // batch_size × params.PublicKeyBytes
+    secretKeys core.HostOrDeviceSlice,  // batch_size × params.SecretKeyBytes
+) runtime.EIcicleError
+```
+
+### Encapsulation
+
+```go
+func Encapsulate(
+    params *KyberParams,
+    message core.HostOrDeviceSlice,     // batch_size × 32 bytes
+    publicKeys core.HostOrDeviceSlice,  // batch_size × params.PublicKeyBytes
+    config MlKemConfig,
+    ciphertexts core.HostOrDeviceSlice,  // batch_size × params.CiphertextBytes
+    sharedSecrets core.HostOrDeviceSlice, // batch_size × params.SharedSecretBytes
+) runtime.EIcicleError
+```
+
+### Decapsulation
+
+```go
+func Decapsulate(
+    params *KyberParams,
+    secretKeys core.HostOrDeviceSlice,    // batch_size × params.SecretKeyBytes
+    ciphertexts core.HostOrDeviceSlice,   // batch_size × params.CiphertextBytes
+    config MlKemConfig,
+    sharedSecrets core.HostOrDeviceSlice, // batch_size × params.SharedSecretBytes
+) runtime.EIcicleError
+```
+
+All buffers may live either on the **host** or on the currently-active **device**.
+
+### `MlKemConfig`
+
+```go
+type MlKemConfig struct {
+    StreamHandle          runtime.Stream
+    IsAsync               bool
+    MessagesOnDevice      bool
+    EntropyOnDevice       bool
+    PublicKeysOnDevice    bool
+    SecretKeysOnDevice    bool
+    CiphertextsOnDevice   bool
+    SharedSecretsOnDevice bool
+    BatchSize             uint64
+    Ext                   config_extension.ConfigExtensionHandler
+}
+```
+
+Use `GetDefaultMlKemConfig()` to get the default configuration. Setting `IsAsync = true` lets the call return immediately; remember to synchronize the stream before reading results.
+
+### Kyber Parameters
+
+The package supports three parameter sets:
+
+- **`Kyber512Params`**: Security level 1 (128-bit security)
+- **`Kyber768Params`**: Security level 3 (192-bit security)
+- **`Kyber1024Params`**: Security level 5 (256-bit security)
+
+```go
+type KyberParams struct {
+    PublicKeyBytes    int
+    SecretKeyBytes    int
+    CiphertextBytes   int
+    SharedSecretBytes int
+    K                 uint8
+    Eta1              uint8
+    Eta2              uint8
+    Du                uint8
+    Dv                uint8
+    // ... internal FFI functions
+}
+```
+
+Constants:
+
+- `ENTROPY_BYTES = 64` - entropy required per keypair generation
+- `MESSAGE_BYTES = 32` - message size for encapsulation
+
+---
+
+## Quick-start example (Kyber768, host buffers)
```

Comment:
We decided not to include full examples in the docs as it is difficult to maintain and update. Please remove this and link to the example here.